src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -193,7 +193,15 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
 
     // Method Description:
     // - Loads the search box from the xaml UI and focuses it.
-    void TermControl::CreateSearchBoxControl()
+    // Arguments:
+    // - populateFromSelection: if this flag set to true and there is a text is selected inside terminal,
+    // we should try to use the selected text to populate the search box.
+    // - allowMultiLineSelection: if this flag set to true and multiple lines are selected,
+    // we should use all the lines to populate the search box.
+    // If the flag is set to false and multiple lines are selected,
+    // we should not use the selected text to populate the search box.
+    // - caseSensitive: boolean that represents if the current search is case sensitive
+    void TermControl::CreateSearchBoxControl(bool populateFromSelection, bool allowMultiLineSelection)

```

Comment:
Personally not a fan of `allowMultiLineSelection` only being used if `populateFromSelection` is true. Why not pass in the `FindSelectionMode` directly and do these checks in here?

Reply:
I believe that TermControl is not aware of the model, and I didn't want to change this.

Reply:
Fair enough. Resolving this suggestion then :)